### PR TITLE
Dedicated flows for SP based IAL2 flows

### PR DESCRIPTION
**Pull request by jgrevich**
This creates dedicated flows for SP based IAL2 sign in/up which can be used with the `id-locust` helper script. Currently the only way to do this is via the prod_simulator flow.

sp_ial2_sign_up
```
# id-locust sp_ial2_sign_up pt -u 10 -n 10000 -t 15m 
+ locust --host https://idp.pt.identitysandbox.gov --locustfile /etc/login.gov/repos/identity-loadtest/load_testing/sp_ial2_sign_up.locustfile.py --users 10 --spawn-rate 1 --headless --csv /var/log/loadtest/20210727194640-sp_ial2_sign_up --run-time 15m
/usr/lib/python3/dist-packages/requests/__init__.py:80: RequestsDependencyWarning: urllib3 (1.26.6) or chardet (3.0.4) doesn't match a supported version!
  RequestsDependencyWarning)
[2021-07-27 19:46:41,307] jumphost-i-030fc47cf42e30503.pt.identitysandbox.gov/INFO/locust.main: Run time limit set to 900 seconds
[2021-07-27 19:46:41,307] jumphost-i-030fc47cf42e30503.pt.identitysandbox.gov/INFO/locust.main: Starting Locust 1.6.0
[2021-07-27 19:46:41,308] jumphost-i-030fc47cf42e30503.pt.identitysandbox.gov/INFO/locust.runners: Spawning 10 users at the rate 1 users/s (0 users already running)...
...
 Name                                                          # reqs      # fails  |     Avg     Min     Max  Median  |   req/s failures/s
--------------------------------------------------------------------------------------------------------------------------------------------
 POST /login/two_factor/sms                                       352     0(0.00%)  |     338     288     456     330  |    0.60    0.00
 GET /phone_setup                                                 353     0(0.00%)  |      73      58     143      69  |    0.60    0.00
 POST /phone_setup                                                353     0(0.00%)  |     232     189     322     230  |    0.60    0.00
 POST /sign_up/completed                                          343     0(0.00%)  |     401     314     618     400  |    0.30    0.00
 POST /sign_up/create_password                                    353     0(0.00%)  |     349     317     436     350  |    0.50    0.00
 GET /sign_up/email/confirm?confirmation_token=                   353     0(0.00%)  |     118      94     356     110  |    0.40    0.00
 GET /sign_up/enter_email                                         353     0(0.00%)  |      60      43    1107      53  |    0.40    0.00
 POST /sign_up/enter_email                                        353     0(0.00%)  |     295     245     469     280  |    0.40    0.00
 POST /verify/confirmations                                       344     0(0.00%)  |     202     158     297     200  |    0.30    0.00
 PUT /verify/doc_auth/agreement                                   352     0(0.00%)  |     161     127     309     160  |    0.60    0.00
 PUT /verify/doc_auth/document_capture                            350     2(0.57%)  |    8732    8360   58236    8400  |    0.40    0.00
 PUT /verify/doc_auth/ssn                                         348     0(0.00%)  |     165     133     287     160  |    0.40    0.00
 PUT /verify/doc_auth/upload?type=desktop                         352     0(0.00%)  |     164     137     272     160  |    0.60    0.00
 PUT /verify/doc_auth/verify                                      344     0(0.00%)  |    3046    2893    3970    3000  |    0.30    0.00
 PUT /verify/doc_auth/welcome                                     352     0(0.00%)  |     163     131     509     160  |    0.60    0.00
 PUT /verify/otp_delivery_method                                  344     0(0.00%)  |     238     190     335     240  |    0.30    0.00
 PUT /verify/phone                                                344     0(0.00%)  |    1464    1378    1618    1500  |    0.30    0.00
 PUT /verify/phone_confirmation                                   344     0(0.00%)  |     229     183     386     230  |    0.30    0.00
 PUT /verify/review                                               344     0(0.00%)  |    1849    1751    2062    1800  |    0.30    0.00
 GET https://idp.pt.identitysandbox.gov/openid_connect/logout     343     0(0.00%)  |      83      63     146      79  |    0.30    0.00
 GET https://sp-oidc-sinatra.pt.identitysandbox.gov               353     0(0.00%)  |       5       3      44       5  |    0.40    0.00
 GET https://sp-oidc-sinatra.pt.identitysandbox.gov/auth/request?aal=&ial=2     353     0(0.00%)  |     107      82     212     100  |    0.40    0.00
--------------------------------------------------------------------------------------------------------------------------------------------
 Aggregated                                                      7680     2(0.03%)  |     837       3   58236     210  |    9.30    0.00
```

sp_ial2_sign_in
```
# id-locust sp_ial2_sign_in pt -u 10 -n 10000 -t 15m 
+ locust --host https://idp.pt.identitysandbox.gov --locustfile /etc/login.gov/repos/identity-loadtest/load_testing/sp_ial2_sign_in.locustfile.py --users 10 --spawn-rate 1 --headless --csv /var/log/loadtest/20210727200243-sp_ial2_sign_in --run-time 15m
/usr/lib/python3/dist-packages/requests/__init__.py:80: RequestsDependencyWarning: urllib3 (1.26.6) or chardet (3.0.4) doesn't match a supported version!
  RequestsDependencyWarning)
[2021-07-27 20:02:43,680] jumphost-i-030fc47cf42e30503.pt.identitysandbox.gov/INFO/locust.main: Run time limit set to 900 seconds
[2021-07-27 20:02:43,680] jumphost-i-030fc47cf42e30503.pt.identitysandbox.gov/INFO/locust.main: Starting Locust 1.6.0
[2021-07-27 20:02:43,680] jumphost-i-030fc47cf42e30503.pt.identitysandbox.gov/INFO/locust.runners: Spawning 10 users at the rate 1 users/s (0 users already running)...
...
 Name                                                          # reqs      # fails  |     Avg     Min     Max  Median  |   req/s failures/s
--------------------------------------------------------------------------------------------------------------------------------------------
 POST /                                                           109     0(0.00%)  |     692     586     937     650  |    0.60    0.00
 POST /login/two_factor/sms                                       109     3(2.75%)  |     327     275     469     320  |    0.60    0.10
 POST /sign_up/completed                                           99     0(0.00%)  |     509     338    7623     390  |    0.50    0.00
 POST /verify/confirmations                                        99     0(0.00%)  |     204     173     277     200  |    0.50    0.00
 PUT /verify/doc_auth/agreement                                   104     0(0.00%)  |     162     135     250     160  |    0.50    0.00
 PUT /verify/doc_auth/document_capture                            100     0(0.00%)  |    8574    8334   18689    8400  |    0.10    0.00
 PUT /verify/doc_auth/ssn                                         100     0(0.00%)  |     167     135     252     160  |    0.10    0.00
 PUT /verify/doc_auth/upload?type=desktop                         104     0(0.00%)  |     165     134     241     160  |    0.50    0.00
 PUT /verify/doc_auth/verify                                      100     0(0.00%)  |    3020    2878    6044    3000  |    0.30    0.00
 PUT /verify/doc_auth/welcome                                     104     0(0.00%)  |     162     134     209     160  |    0.50    0.00
 PUT /verify/otp_delivery_method                                   99     0(0.00%)  |     211     168     294     210  |    0.40    0.00
 PUT /verify/phone                                                 99     0(0.00%)  |    1448    1356    4470    1400  |    0.40    0.00
 PUT /verify/phone_confirmation                                    99     0(0.00%)  |     218     174     339     210  |    0.40    0.00
 PUT /verify/review                                                99     0(0.00%)  |    1860    1768    2042    1800  |    0.50    0.00
 GET https://idp.pt.identitysandbox.gov/openid_connect/logout      99     0(0.00%)  |      83      67     130      80  |    0.50    0.00
 GET https://sp-oidc-sinatra.pt.identitysandbox.gov               110     0(0.00%)  |       6       3      52       5  |    0.50    0.00
 GET https://sp-oidc-sinatra.pt.identitysandbox.gov/auth/request?aal=&ial=2     110     0(0.00%)  |     110      88     189     110  |    0.50    0.00
--------------------------------------------------------------------------------------------------------------------------------------------
 Aggregated                                                      1743     3(0.17%)  |    1033       3   18689     210  |    7.40    0.10
```

### CODE REVIEW

File: load_testing/sp_ial2_sign_up.locustfile.py [hunk 1]
```diff
@@ -0,0 +1,17 @@
+from locust import HttpUser, TaskSet, task, between
+from common_flows import flow_sp_ial2_sign_up, flow_helper
+
+
+class SP_IAL2_SignUpLoad(TaskSet):
+    # Preload drivers license data
+    license_front = flow_helper.load_fixture("mont-front.jpeg")
+    license_back = flow_helper.load_fixture("mont-back.jpeg")
```
Comment (zachmargolis) [inline load_testing/sp_ial2_sign_up.locustfile.py:8]: where are these used again?

Comment (jgrevich) [reply load_testing/sp_ial2_sign_up.locustfile.py:8]: Those are used during the doc auth step here: https://github.com/18F/identity-loadtest/blob/main/load_testing/common_flows/flow_sp_ial2_sign_in.py#L144

Comment (zachmargolis) [reply load_testing/sp_ial2_sign_up.locustfile.py:8]: Oh weird, I didn't realize that class variables would be loaded on the instance like that, neat

**Review by zachmargolis — APPROVED** [review]
LGTM

File: load_testing/sp_ial2_sign_up.locustfile.py (hunk 1 above)
Comment (zachmargolis) [reply load_testing/sp_ial2_sign_up.locustfile.py:8]: Made a lil test to confirm:

```python
class Foobar:
  license_front = 1


  def run(self):
    print(self.license_front)

x = Foobar()
x.run()
# prints 1
```

Comment (jgrevich) [reply load_testing/sp_ial2_sign_up.locustfile.py:8]: I didn't either, I've been having a lot of fun learning python

**Pull request merged**